src/libraries/System.Runtime.InteropServices/tests/LibraryImportGenerator.UnitTests/CompileFails.cs

```diff
@@ -22,6 +22,31 @@ namespace LibraryImportGenerator.UnitTests
 {
     public class CompileFails
     {
+        [Fact]
+        public Task GenericDelegateParameterFails()
+        {
+            var src = """
+            using System;
+            using System.Runtime.InteropServices;
+
+            partial class Test
+            {
+                [LibraryImportAttribute("DoesNotExist")]
+                public static partial void Method1(Func<int, int> {|#0:f|});
+            }
+            """;
+
+
+            return VerifyCS.VerifySourceGeneratorAsync(
+                src,
+                new DiagnosticResult[]
+                {
+                    VerifyCS.Diagnostic(GeneratorDiagnostics.ParameterTypeNotSupportedWithDetails)
+                        .WithLocation(0)
+                        .WithArguments("Marshalling a generic delegate is not supported. Consider using a function pointer instead.", "f")
+                });
+        }
+
```

Comment:
I know I said that I'd be fine with a custom test here, but seeing what you're writing now, can you move this test to use `CodeSnippets.BasicParametersAndModifiers("System.Func<int, int>")` for consistency with the other cases we have below? That way we test all the different types of parameter passing as well as return types.